application.go

```diff
@@ -523,6 +542,15 @@ func applicationV3Fields() (schema.Fields, schema.Defaults) {
 	return fields, defaults
 }
 
+func applicationV4Fields() (schema.Fields, schema.Defaults) {
+	fields, defaults := applicationV3Fields()
+	fields["placement"] = schema.String()
+	fields["desired-scale"] = schema.Int()
+	defaults["placement"] = ""
+	defaults["desired-scale"] = int64(0)
```

Comment:
I am not sure why int64 is needed here sine you have declared DesiredScale as int...

Reply:
Copying what was done originally for min units. I suspect it's so that it's robust to serialisation which tends to use int64.